lib/snippets/deploy-to-gke.rb

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env ruby
+
+# Usage: deploy-to-gke <app's namespace> <gcloud deployment> <gcloud project>
+
+# Requires the following to be available on Shipit machine:
+#  - gcloud binary available on the shipit machine's path
+#  - kubectl binary available in the shipit machine's path
+#  - GCLOUD_CREDENTIALS_DIR: location of JSON service account auth files, e.g. /home/username/.google_iam
+
+# Optionally, the following variables can be used to override script defaults:
+#  - K8S_TEMPLATE_FOLDER: location of Kubernetes files to deploy. Default is config/<environment>/.
+
+require 'open3'
+require 'securerandom'
+require 'json'
+require 'pry'
+
+class GKEDeployment
+  class FatalDeploymentError < StandardError; end
+
+  def initialize(namespace:, gcloud_deployment:, gcloud_project:, environment:, current_sha:, template_folder: nil)
+    @namespace = namespace
+    @gcloud_deployment = gcloud_deployment
+    @gcloud_project = gcloud_project
+    @current_sha = current_sha
+    enforce_required_parmas
+    @id = current_sha + "-#{SecureRandom.hex(4)}"
+    @template_path = template_folder || "config/kubernetes/#{environment}/"
```

Comment:
(also for line 11) The default I went with in my other PR is `config/deploy/#{env}`. We can change that if you think this is better, but we should choose the same thing.